megamek/src/megamek/utils/MegaMekXmlUtil.java

```diff
@@ -129,94 +139,321 @@ public static XMLReader createSafeXMLReader() {
+    /**
+     * This writes an open XML tag, with the possible addition of a class attribute and its class
+     * @param pw the PrintWriter to use
+     * @param indent the indent to write at
+     * @param name the name of the XML tag
+     * @param classAttribute the class attribute to write as part of the tag
+     * @param c the class to write as part of the tag
+     */
+    public static void writeSimpleXMLOpenTag(final PrintWriter pw, final int indent,
+                                             final String name, final String classAttribute,
+                                             final Class<?> c) {
+        writeSimpleXMLOpenTag(pw, indent, name, null, null, classAttribute, c);
+    }
+
+    /**
+     * This writes an open XML tag, with the possible addition of an attribute and its value
+     * and/or the possible addition of a class attribute and its class
+     * @param pw the PrintWriter to use
+     * @param indent the indent to write at
+     * @param name the name of the XML tag
+     * @param attribute the attribute to write as part of the XML tag
+     * @param value the value of the attribute
+     * @param classAttribute the class attribute to write as part of the tag
+     * @param c the class to write as part of the tag
+     */
+    public static <T> void writeSimpleXMLOpenTag(final PrintWriter pw, final int indent,
+                                                 final String name,
+                                                 final @Nullable String attribute,
+                                                 final @Nullable T value,
+                                                 final @Nullable String classAttribute,
+                                                 final @Nullable Class<?> c) {
+        final boolean hasValue = value != null;
+        final boolean hasClass = c != null;
+        pw.print(indentStr(indent) + "<" + name);
+        if (hasValue) {
+            pw.print(" " + attribute + "=\"" + value + "\"");
         }
+
+        if (hasClass) {
+            pw.print(" " + classAttribute + "=\"" + c.getName() + "\"");
+        }
+        pw.print(">\n");
     }
+    //endregion Open Tag
 
-    public static void writeSimpleXmlTag(PrintWriter pw1, int indent, String name, String val) {
-        pw1.println(indentStr(indent) + "<" + name + ">" + escape(val) + "</" + name + ">");
+    //region Simple XML Tag
+    /**
+     * This writes a UUID or an array of UUIDs to file
+     * @param pw the PrintWriter to use
+     * @param indent the indent to write at
+     * @param name the name of the XML tag
+     * @param values the UUID or UUID[] to write to XML
+     */
+    public static void writeSimpleXMLTag(final PrintWriter pw, final int indent, final String name,
+                                         final UUID... values) {
+        if (values.length > 0) {
+            final StringJoiner stringJoiner = new StringJoiner(",", "", "");
```

Comment:
Is this constructor somehow different than the simpler, `new StringJoiner(",")` ?

Reply:
Just swapped it over... and it is working fine with that constructor now. I was using the three terms because that's the only way my IDE didn't give me an error